perfkitbenchmarker/providers/ibmcloud/ibmcloud_disk.py

```diff
@@ -0,0 +1,199 @@
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module containing classes related to IBM Cloud disks """
+
+import json
+import logging
+import time
+import re
+
+from absl import flags
+from perfkitbenchmarker import disk
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import vm_util
+
+from perfkitbenchmarker.providers.ibmcloud import ibm_api as ibm
+
+FLAGS = flags.FLAGS
+
+ATTACHED = 'attached'
+AVAILABLE = 'available'
+DEFAULT_DISK_SIZE = 1000
+
+
+class IbmCloudDisk(disk.BaseDisk):
+  """Object representing a IBM Cloud Disk"""
+
+  def __init__(self, disk_spec, name, zone):
+    super(IbmCloudDisk, self).__init__(disk_spec)
+    self.name = name
+    self.zone = zone
+    self.vol_id = None
+    self.attached_vm = None
+    self.attached_vdisk_URI = None
+    self.device_path = None
+    self.data_disk_size = FLAGS.data_disk_size if FLAGS.data_disk_size is not None else DEFAULT_DISK_SIZE
+
+  def _Create(self):
+    """Creates an external block volume."""
+    volcmd = ibm.IbmAPICommand(self)
+    volcmd.flags['name'] = self.name
+    volcmd.flags['zone'] = self.zone
+    volcmd.flags['iops'] = FLAGS.ibmcloud_volume_iops
+    volcmd.flags['profile'] = FLAGS.ibmcloud_volume_profile
+    volcmd.flags['capacity'] = self.data_disk_size
+    if FLAGS.ibmcloud_datavol_encryption_key:
+      volcmd.flags['encryption_key'] = FLAGS.ibmcloud_datavol_encryption_key
+    if FLAGS.ibmcloud_rgid:
+      volcmd.flags['resource_group'] = FLAGS.ibmcloud_rgid
+
+    logging.info('Volume create, volcmd.flags %s', volcmd.flags)
+    resp = json.loads(volcmd.CreateVolume())
+    self.vol_id = resp['id']
+    logging.info('Volume id: %s', self.vol_id)
+
+  def _Delete(self):
+    """Deletes an external block volume."""
+    if self.vol_id is None:
+      logging.info('Volume %s was not created. Skipping deletion.' % self.name)
+      return
+    volcmd = ibm.IbmAPICommand(self)
+    volcmd.flags['volume'] = self.vol_id
+    logging.info('Volume delete, volcmd.flags %s', volcmd.flags)
+    volcmd.DeleteVolume()
+    logging.info('Volume deleted: %s', self.vol_id)
+    self.vol_id = None
+
+  def _Exists(self):
+    if self.vol_id is None:
+      return False
+    return True
+
+  def _FindVdiskIdByName(self, vdisks, name):
+    """Finds vdisk id by name"""
+    for vdisk in vdisks:
+      if vdisk['name'] == name:
+        return vdisk['id']
+    return None
+
+  def Attach(self, vm):
+    """Attaches the disk to a VM.
+
+    Args:
+      vm: instance of the vm to which the disk will be attached.
+    """
+    if self.vol_id is None:
+      raise errors.Error('Cannot attach remote volume %s' % self.name)
```

Comment:
Can you use f-strings here?
  f'Cannot attach remote volume {self.name}'

Reply:
changed to f string.